test/skate_web/channels/detours_channel_test.exs

```diff
@@ -112,13 +91,21 @@ defmodule SkateWeb.DetoursChannelTest do
     test "subscribes to draft detours with initial detours", %{conn: conn, socket: socket} do
       %{id: authenticated_user_id} = SkateWeb.AuthManager.Plug.current_resource(conn)
 
+      :detour |> build() |> with_id(1) |> insert()
+      :detour |> build() |> with_id(2) |> insert()
+      :detour |> build() |> with_id(3) |> insert()
+
+      put(conn, ~p"/api/detours/update_snapshot", %{
+        "snapshot" => :detour_snapshot |> build() |> with_id(4)
+      })
```

Comment:
question: can you explain why this can't use the `:detour |> [...] |> insert()` method?

Reply:
I see! I think this should work
https://github.com/mbta/skate/pull/2908/files/8acf7072c0392401f45e97dfc4e47d1e83b134a8..ecd135c15c69f5510318f45962d851c35238520e#r1910687982
```suggestion
    :detour |> build(user: user) |> with_id(4)
```

Reply:
OH BEAUTIFUL! Yes it sure does work, thank you 🚀